auth/common.hh

```diff
@@ -84,14 +84,13 @@ future<> create_legacy_metadata_table_if_missing(
 // Use this function when need to perform read before write on a single guard or if
 // you have more than one mutation and potentially exceed single command size limit.
 using start_operation_func_t = std::function<future<::service::group0_guard>(abort_source*)>;
-using mutations_generator = coroutine::experimental::generator<mutation>;
```

Comment:
This patch could just disappear, no? By using the other mutations_generator in the first place.

Reply:
yes, it was in a single commit before, but Piotr was complaining about too big commits so I split it.

code in auth/common couldn't use code from raft_group0_client right away because it's on master already.